controller_manager:
  ros__parameters:
    update_rate: 100
    use_sim_time: true

    diff_drive_controller:
      type: diff_drive_controller/DiffDriveController
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    shoulder_yaw_joint_controller:
      type: position_controllers/JointPositionController
    shoulder_pitch_joint_controller:
      type: position_controllers/JointPositionController
    elbow_joint_controller:
      type: position_controllers/JointPositionController
    wrist_roll_joint_controller:
      type: position_controllers/JointPositionController
    wrist_pitch_joint_controller:
      type: position_controllers/JointPositionController
    gripper_controller:
      type: position_controllers/JointGroupPositionController

diff_drive_controller:
  ros__parameters:
    use_sim_time: true
    left_wheel_names: ["front_left_wheel_joint", "rear_left_wheel_joint"]
    right_wheel_names: ["front_right_wheel_joint", "rear_right_wheel_joint"]
    wheel_separation: 0.55
    wheel_radius: 0.1
    publish_rate: 50.0
    odom_frame_id: odom
    base_frame_id: base_footprint
    enable_odom_tf: true
    cmd_vel_timeout: 0.5

shoulder_yaw_joint_controller:
  ros__parameters:
    joints:
      - shoulder_yaw_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

shoulder_pitch_joint_controller:
  ros__parameters:
    joints:
      - shoulder_pitch_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

elbow_joint_controller:
  ros__parameters:
    joints:
      - elbow_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

wrist_roll_joint_controller:
  ros__parameters:
    joints:
      - wrist_roll_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

wrist_pitch_joint_controller:
  ros__parameters:
    joints:
      - wrist_pitch_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity

gripper_controller:
  ros__parameters:
    joints:
      - left_finger_joint
      - right_finger_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity